File: Poin A/PC.cpp
#include<iostream>

using namespace std;

#include "Processor.cpp"
#include "Disk.cpp"
#include "RAM.cpp"

class PC{
    private :
        //atribut
        Processor p;
        Disk d;
        RAM r;
        int totalPrice;

    public :
        //konstruktor
        PC(){}

        PC(Processor p, Disk d, RAM r){
            this->p = p;
            this->d = d;
            this->r = r;
            this->totalPrice = p.getPriceP() + d.getPriceD() + r.getPriceR();
            
        }

        //setter & getter
        void setProcessor(Processor p){
            this->p = p;
        }

        void setDisk(Disk d){
            this->d = d;
        }

        void setRAM(RAM r){
            this->r = r;
        }
        void setTotalPrice(Processor p, Disk d, RAM r){
             this->totalPrice = p.getPriceP() + d.getPriceD() + r.getPriceR();
        }
        
        Processor getProcessor(){
            return this->p;
        }

        Disk getDisk(){
            return this->d;
        }

        RAM getRAM(){
            return this->r;

        }

        
        int getTotalPrice(){
            return this->totalPrice;
        }

        



        //output
        void printPC(){
            p.printProcessor();
            d.printType();
            r.printRAM();
            cout << "------------------------------------" << endl;
            cout << "Total Harga  : " << getTotalPrice() << endl;
            cout << "====================================" << endl;

        }

        ~PC(){}

};

File: Poin B.java/PC.java
public class PC {
    private Processor p;
    private Disk d;
    private RAM r;
    private int totalPrice;


    
    public PC(Processor p, Disk d, RAM r) {
        this.p = p;
        this.d = d;
        this.r = r;
        this.totalPrice = p.getPrice() + d.getPrice() + r.getPrice();

    }
    public int getTotalPrice() {
        return totalPrice;
    }
    public void setTotalPrice(Processor p, Disk d, RAM r) {
        this.totalPrice = p.getPrice() + d.getPrice() + r.getPrice();
    }
    public PC() {
    }
    public RAM getR() {
        return r;
    }
    public Processor getP() {
        return p;
    }
    public void setP(Processor p) {
        this.p = p;
    }
    public Disk getD() {
        return d;
    }
    public void setD(Disk d) {
        this.d = d;
    }
    public void setR(RAM r) {
        this.r = r;
    }

    void output(){
        p.output();
        d.output();
        r.output();
        System.out.println("-------------------------------------+");
        System.out.println("Total Price             : " + this.getTotalPrice());
    }


}
